cpp/src/interop/from_arrow_host.cu

```diff
@@ -53,119 +53,193 @@ namespace detail {
 
 namespace {
 
+/**
+ * @brief Return bitmask word at the given index in the source
+ *
+ * This is a 64-bit version of cudf::detail::get_mask_offset_word
+ * since the source may have a range more than max(int) bits.
+ */
```

Comment:
Silly question: If the main difference between these is the data-type, would there be value in combining the functions into a function template, typed on index_type?